azure-storage-queue/src/main/scala/akka/stream/alpakka/azure/storagequeue/javadsl/AzureQueueSink.scala

```diff
@@ -44,27 +40,21 @@ object AzureQueueWithTimeoutsSink {
    * This is the same as [[AzureQueueSink.create]] expect that it takes instead
    * of a [[com.microsoft.azure.storage.queue.CouldQueueMessage]] a [[MessageWithTimeouts]].
    */
-  def create(cloudQueue: CloudQueue, maxInFlight: Int): Sink[MessageWithTimeouts, CompletionStage[Done]] =
+  def create(cloudQueue: Supplier[CloudQueue]): Sink[MessageWithTimeouts, CompletionStage[Done]] =
     AzureQueueSink.fromFunction(
-      (input: MessageWithTimeouts, ec: ExecutionContext) =>
-        AzureQueueSinkFunctions.addMessage(cloudQueue)(input.message, input.timeToLive, input.initialVisibility)(ec),
-      maxInFlight
+      { input: MessageWithTimeouts =>
+        AzureQueueSinkFunctions.addMessage(cloudQueue.get())(input.message, input.timeToLive, input.initialVisibility)
+      }
```

Comment:
`fromFunction` expects the function to be non-blocking, so use the blocking IO dispatcher through attributes here as well, something like this: `Flow.fromFunction(...).addAttributes(Attributes(DefaultAttributes.IODispatcher))`